modules/subnet/subnet.tf

```diff
@@ -10,6 +10,8 @@ resource "oci_core_subnet" "vcn_subnet" {
   cidr_block     = each.value.cidr_block
   compartment_id = var.compartment_id
   vcn_id         = var.vcn_id
+  availability_domain = lookup(each.value, "availability_domain", null)
```

Comment:
Please avoid null if possible.

Reply:
null cannot be avoided as it is required to handle availability domain specific subnets as well as regional subnets.
As per documentation (https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_subnet), for creating availability domain specific subnet, availability domain is required and for regional subnets, availability domain is required to be skipped which needs to be handled via null.

Reply:
In this case, I think you should have a default value.

Reply:
@hyder - Can you suggest what default value are you expecting? The use case here is to handle local ad specific and regional subnets which require null for creating regional subnets. I checked the link https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/module-iam.tf and its not applicable for this use case.

Reply:
Most regions have 1 AD. So perhaps AD 1? @Djelibeybi @devoncrouse what do you think?

Reply:
@Djelibeybi  @devoncrouse  - Can you please comment when time permits ?